File: Professional Image Modifier 5/[C#]-Professional Image Modifier 5/C#/ImageFunctions/Classes/Utilities.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageFunctions.Classes
{
    class Utilities
    {

        public static String BytesToString(long byteCount)
        {
            string[] suf = { "B", "KB", "MB", "GB", "TB", "PB", "EB" }; //Longs run out around EB
            if (byteCount == 0)
                return "0" + suf[0];
            long bytes = Math.Abs(byteCount);
            int place = Convert.ToInt32(Math.Floor(Math.Log(bytes, 1024)));
            double num = Math.Round(bytes / Math.Pow(1024, place), 1);
            return (Math.Sign(byteCount) * num).ToString() + suf[place];
        }

        // Convert byte[][] to string
        public static string ConvertToString(byte[][] p)
        {
            StringBuilder sb = new StringBuilder();
            foreach (byte[] b in p)
            {
                sb.Append(System.Text.Encoding.Default.GetString(b));
            }
            return sb.ToString();
        }
    }
}

File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Classes/Utilities.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ImageFunctions.Classes
	Friend Class Utilities

		Public Shared Function BytesToString(ByVal byteCount As Long) As String
			Dim suf() As String = { "B", "KB", "MB", "GB", "TB", "PB", "EB" } 'Longs run out around EB
			If byteCount = 0 Then
				Return "0" & suf(0)
			End If
			Dim bytes As Long = Math.Abs(byteCount)
			Dim place As Integer = Convert.ToInt32(Math.Floor(Math.Log(bytes, 1024)))
			Dim num As Double = Math.Round(bytes / Math.Pow(1024, place), 1)
			Return (Math.Sign(byteCount) * num).ToString() & suf(place)
		End Function

		' Convert byte[][] to string
		Public Shared Function ConvertToString(ByVal p()() As Byte) As String
			Dim sb As New StringBuilder()
			For Each b As Byte() In p
				sb.Append(System.Text.Encoding.Default.GetString(b))
			Next b
			Return sb.ToString()
		End Function
	End Class
End Namespace
